tests/unit/test_charm.py

```diff
@@ -1,5 +1,5 @@
 import unittest
-from unittest.mock import PropertyMock, patch
+from unittest.mock import MagicMock, PropertyMock, patch
```

Comment:
Why are you importing `MagicMock` here? It does not look like you use it anywhere throughout your tests.

`PropertyMock` is good though. I started using it in my tests too :smile:!

Reply:
Oh weird, I thought I removed that good catch will remove